tapchannel/aux_invoice_manager.go

```diff
@@ -227,6 +238,64 @@ func (s *AuxInvoiceManager) priceFromQuote(rfqID rfqmsg.ID) (
 	}
 }
 
+// rfqPeerFromScid attempts to match the provided scid with a negotiated quote,
+// then it returns the RFQ peer's node id.
+func (s *AuxInvoiceManager) rfqPeerFromScid(scid uint64) (route.Vertex, error) {
+	acceptedBuyQuotes := s.cfg.RfqManager.PeerAcceptedBuyQuotes()
+	acceptedSellQuotes := s.cfg.RfqManager.LocalAcceptedSellQuotes()
+
+	buyQuote, isBuy := acceptedBuyQuotes[rfqmsg.SerialisedScid(scid)]
+	sellQuote, isSell := acceptedSellQuotes[rfqmsg.SerialisedScid(scid)]
+
+	switch {
+	case isBuy:
```

Comment:
Still unaddressed. If there's no good reason why we'd eve want to check a sell order, I think we should change this to only look at buy orders.

Reply:
Turns out I was wrong here! In case of a direct-peer payment through an asset invoice, it can actually be a sell order SCID. The itest failed here, so will update it.